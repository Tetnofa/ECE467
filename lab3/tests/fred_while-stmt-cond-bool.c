int f1() {
  int a;
  bool b;
  while (1) {
  }
  while (true) {
  }
  while (true) {
  }
  while (a + a) {
  }
  while (a == a && b) {
  }
}
