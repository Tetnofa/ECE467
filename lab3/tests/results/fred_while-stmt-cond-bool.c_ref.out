Program [useIO=0](1,0)
	Function[isProto=0]	Int(1,0)
		Identifier[name:f1](1,4)
		Scope(1,9)
			Scalar Declaration			Int(2,2)
				Identifier[name:a](2,6)
			Scalar Declaration			Bool(3,2)
				Identifier[name:b](3,7)
			WhileStmt(4,2)
				Int Expression(4,9)
					IntConstant[val=1](4,9)
				Scope(4,12)
			WhileStmt(6,2)
				Int Expression(6,9)
					BoolConstant[val=1](6,9)
				Scope(6,15)
			WhileStmt(8,2)
				Int Expression(8,9)
					BoolConstant[val=1](8,9)
				Scope(8,15)
			WhileStmt(10,2)
				Int Expression(10,9)
					Binary Expression[opcode:+](10,9)
						Int Expression(10,9)
							Reference(10,9)
								Identifier[name:a](10,9)
						Int Expression(10,13)
							Reference(10,13)
								Identifier[name:a](10,13)
				Scope(10,16)
			WhileStmt(12,2)
				Bool Expression(12,9)
					Binary Expression[opcode:&&](12,9)
						Bool Expression(12,9)
							Binary Expression[opcode:==](12,9)
								Int Expression(12,9)
									Reference(12,9)
										Identifier[name:a](12,9)
								Int Expression(12,14)
									Reference(12,14)
										Identifier[name:a](12,14)
						Int Expression(12,19)
							Reference(12,19)
								Identifier[name:b](12,19)
				Scope(12,22)
sema: 4:2 : invalid condition in while statement
sema: 10:2 : invalid condition in while statement
